guests/python/src/python_modules/mod.rs

```diff
@@ -0,0 +1,1527 @@
+//! Python modules that are injected by us.
+use pyo3::{BoundObject, exceptions::PyValueError, prelude::*};
+
+mod error;
+
+use error::{ResourceMoved, ResourceMovedOptionExt};
+
+/// Register python modules.
+///
+/// # Panic
+/// This must be called BEFORE the interpreter is used.
+pub(crate) fn register() {
+    pyo3::append_to_inittab!(wit_world);
+}
+
+/// Provide a [`componentize-py`]-compatible Python API.
```

Comment:
for the reviewers: this here explains the implementation and some broader ecosystem considerations.